keras/layers/core/einsum_dense.py

```diff
@@ -197,6 +197,10 @@ def kernel(self):
             )
         return self._kernel
 
+    @kernel.setter
+    def kernel(self, value):
+        self._kernel.assign(value)
```

Comment:
Is this the intended behavior? This is a value assignment, but would users expect to set the actual variable instead?

Also, should we disable this when LoRA is enabled?

Reply:
I think this should be compatible with lora:

```python
    @kernel.setter
    def kernel(self, value):
        self._kernel.assign(value)
        if self.lora_enabled:
            self.lora_kernel_a.assign(ops.zeros(self.lora_kernel_a.shape))
            self.lora_kernel_b.assign(ops.zeros(self.lora_kernel_b.shape))
```

just like how `load_own_variables` does

Reply:
> This is a value assignment, but would users expect to set the actual variable instead?
Yes, you are right. I changed it to set the actual variable.
The use case I saw was that they try to normalize the kernel during training everytime the layer is called.
I think this use case is OK, but it would be better to set the variable.

For Lora, it would error out even before reaching this setter function. See the new test I added for testing the error.